src/foam/core/IDSupport.js

```diff
@@ -133,8 +140,17 @@ foam.CLASS({
           return prop.clone();
         }),
         methods: [
-          function toString() {
-            return foam.json.Compact.stringify(this, this.cls_);
+          {
+            name: 'toString',
+            code: function() {
+              console.log(this);
+              var arr = [];
+              for ( var prop in this.instance_ ) {
+                arr.push( ! this[prop] ? '' : this[prop].toString() );
```

Comment:
When possible, better to remove the '!' and then reverse the two sides of the '?'.

Reply:
fixed